### ISSUE TITLE
Bash On Windows causes platform.windows_bash_command() to raise CalledProcessError.

### ISSUE THREAD

**Issue by ericmharris** [xonsh/xonsh#2224]
Bash On Ubuntu On Windows (BOW)  is a new feature available in Windows 10 Anniversary Update with Developer Mode enabled.  Background here: https://msdn.microsoft.com/en-us/commandline/wsl/about

Installing BOW creates C:\WINDOWS\system32\bash.exe, and this causes the bash_completer to choke because it's expecting to find git-bash.

I've created a demo fix that's probably a really bad idea, but it'll help show the problem.

https://github.com/ericmharris/xonsh.git

[emharris@eharris-laptop xonsh]# ls Exception in thread Thread-113:
Traceback (most recent call last):
  File "C:\tools\Anaconda3\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "C:\tools\Anaconda3\lib\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "C:\tools\Anaconda3\lib\site-packages\prompt_toolkit\interface.py", line 840, in run
    existing complete_state if we had one. (The user could have
  File "C:\Users\emharris\Dropbox\Development\xonsh\xonsh\ptk\completer.py", line 49, in get_completions
    self.ctx)
  File "C:\Users\emharris\Dropbox\Development\xonsh\xonsh\completer.py", line 35, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "C:\Users\emharris\Dropbox\Development\xonsh\xonsh\completers\bash.py", line 13, in complete_from_bash
    command = xp.bash_command()
  File "C:\Users\emharris\Dropbox\Development\xonsh\xonsh\platform.py", line 329, in bash_command
    bc = windows_bash_command()
  File "C:\Users\emharris\Dropbox\Development\xonsh\xonsh\platform.py", line 309, in windows_bash_command
    universal_newlines=True)
  File "C:\tools\Anaconda3\lib\subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "C:\tools\Anaconda3\lib\subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['C:\\WINDOWS\\system32\\bash.exe', '--version']' returned non-zero exit status 4294967295

**Comment by scopatz** [comment]
Hi @ericmharris, would you mind putting in a PR to help us fix this? I think that would be a better place to continue this discussion, finx you already have a potential fix.

**Comment by scopatz** [comment]
Thanks for reporting!

**Comment by melund** [comment]
Hi @ericmharris. I thought we were already handling this case. I have had a similar issue as well. The Bash from `windows subsystem for linux` (WSL) can't be used for the completer. Thus, we added a check to ensure that were not using that version. 
https://github.com/xonsh/xonsh/blob/master/xonsh/platform.py#L307

On my computer calling the WSL bash doesn't raise any weird errors.
```
C:\Windows\System32>bash.exe --version
GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2013 Free Software Foundation, Inc.
Licens GPLv3+: GNU GPL version 3 eller senere <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

I have added a PR that which catch the subproc error. #2228 

Can you give it a try?

**Comment by scopatz** [comment]
Yes, #2228 has been merged to master, so it would be nice to know if we can close this :) Thanks!

**Comment by scopatz** [comment]
Please reopen if needed

**Comment by ericmharris** [comment]
Hi guys, sorry for ghosting on this but work "shat the bed" ... Fixed on 0.5.7 and thanks for figuring it out.
